src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RegistrationException("Login and password cannot be null");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters.");
+        }
+
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters.");
```

Comment:
The checklist advises against adding something that is not described in the task requirements. If the minimum password length is not a requirement, this check should not be included. If it is a requirement, the value '6' should be extracted to a constant field with a descriptive name.